odl/space/fspace.py

```diff
@@ -379,14 +395,6 @@ def __call__(self, x, out=None, **kwargs):
             if ndim == 1:
                 try:
                     out = self._call(x, **kwargs)
-                    if np.ndim(out) == 0 and not scalar_out:
-                        # Don't accept scalar result. A typical situation where
-                        # this occurs is with comparison operators, e.g.
-                        # "return x > 0" which simply gives 'True' for a
-                        # non-empty tuple (in Python 2). We raise TypeError
-                        # to trigger the call with x[0].
```

Comment:
Update:

It seems we return a 2d array for the 1d case (instead of a tuple as in the higher dimensional cases), this cheeses this problem since the above evaluates as expected.